src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -1411,9 +1413,14 @@ where
             });
         });
 
-        if remaining_space == Some(Sectors(0)) {
+        // It is tempting to exit early if the remaining space is 0. But this
+        // is incorrect. If the remaining space is 0, that just means that the
+        // pool is not overprovisioned and that the total logical size of the
+        // filesystems consumes the whole available space. It is still necessary
+        // to enter the thread scope to detect filesystem usage changes.
```

Comment:
I'm not sure I understand what you mean here. Maybe it's just the placement of the comment, but I was very confused why this wasn't at the point where we would theoretically want to exit early.

Reply:
I'll move it up to where we would exit early.